cmd/conformance-tester/pkg/clients/client_gitops.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
+	"k8c.io/kubermatic/v2/cmd/conformance-tester/pkg/scenarios"
+	ctypes "k8c.io/kubermatic/v2/cmd/conformance-tester/pkg/types"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	kubermaticv1helper "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1/helper"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type gitopsClient struct {
+	opts *ctypes.Options
+}
+
+var _ Client = &gitopsClient{}
+
+func NewGitopsClient(opts *ctypes.Options) Client {
+	return &gitopsClient{
+		opts: opts,
+	}
+}
```

Comment:
Nit: Should this client/runner/mode really be named "gitops"? It seems more accurate to say this is the "CRD" client or the "Kubernetes" client. You don't really need to set up a GitOps pipeline to profit from interacting with the Kubermatic CRDs directly instead of using the Kubermatic API (I'm not opposed to highlight this as "enabling GitOps" in marketing fluff).

Reply:
What I mean is: There's not git involved at all in this client.

Reply:
Tbh, all other names are equally bad in my opinion :D

Reply:
client_kube.go ?

Reply:
I would prefer it to be `client_kube.go` and `NewKubeClient` or `NewKubeCRDClient` or something, yeah. But it's not a hard requirement, I can approve this PR even if it's not renamed.

Reply:
I renamed it.